internal/routes/all_routes.go

```diff
@@ -64,6 +64,9 @@ func apiRouter() http.Handler {
 			r.Route("/{type}", func(r chi.Router) {
 				r.Post("/", s.CreateReservation)
 			})
+			r.Route("/{ID}/status", func(r chi.Router) {
```

Comment:
Can't we directly do:
```suggestion
			r.Get("/{ID}/status", s.GetReservation)
```